src/charm.py

```diff
@@ -1375,10 +1409,30 @@ def _reconciliation(self, _event: EventBase) -> None:
             self._theme_reconciliation()
             self._plugin_reconciliation()
             logger.info("Reconciliation process finished successfully.")
-            self.unit.status = ActiveStatus()
         except exceptions.WordPressStatusException as status_exception:
             logger.info("Reconciliation process terminated early, reason: %s", status_exception)
             self.unit.status = status_exception.status
+            return
+        if self._are_pebble_instances_ready():
+            self.unit.status = ActiveStatus()
+        else:
+            self.unit.status = WaitingStatus("Waiting for pebble")
+
+    def _on_apache_prometheus_exporter_pebble_ready(self, event: PebbleReadyEvent):
+        """Configure and start apache prometheus exporter.
+
+        Args:
+            event: Event triggering the handler.
+        """
+        if not event.workload:
```

Comment:
Out of curiosity, do you know under which scenarios this would occur? I'm guessing the `workload` is the prometheus exporter in this case?

Reply:
An alternative would be to go into blocked state or whatever is appropriate for this event failing

Reply:
Updated using BlockedStatus rather than custom status with indicator saying `Internal Error, pebble container not found`. This case shouldn't really be happening since this event is derived from `WorkloadEvent` which should have container. 
```
class WorkloadEvent(HookEvent):
def __init__(self, handle: 'Handle', workload: 'Container'):
        super().__init__(handle)

        self.workload = workload
```
Thank you for the comment!

Reply:
Perhaps we should raise a bug for the upstream library to fix the type hint if it can't actually be None?

Reply:
Will do :)